compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -2520,7 +2520,8 @@ TR::AMD64RegImm64Instruction::addMetaDataForCodeAddress(uint8_t *cursor)
       {
       if (getNode()->getOpCode().hasSymbolReference() &&
           methodSymRef &&
-          methodSymRef->getReferenceNumber()==TR_referenceArrayCopy)
+          methodSymRef->getReferenceNumber()==TR_referenceArrayCopy ||
+          methodSymRef->getReferenceNumber() == TR_prepareForOSR)
```

Comment:
I tried always permitting methodSymRef relocation, and it leads to crashes,
so it's probably important to only permit relocation for these reference numbers.

Reply:
I'm not sure what you mean by asserting, if the reference number is not known.
All possible reference numbers are not set in one place, despite what the comment implies.
`VMarrayStoreCheckArrayCopyEvaluator` only sets `TR_referenceArrayCopy`.
So I don't know if it would be possible to track all valid reference numbers.

Reply:
Ok, so the answer is "no" :) . I didn't read the code closely enough to realize that "methodSymRef" is really just whatever symbol reference is on the node that caused this code address to be generated. So my "Alternatively" comment is just plain inappropriate.

I guess I just don't, in general, like open ended "if it's one of these special things then do something special" code paths, because such things are easy to become victim to the "I added this new special thing but forgot to add it to the list in *that* spot" which is what this one looks like (you're adding TR_prepareForOSR which wasn't done earlier). So how do we know that other things that need this kind of relocation aren't flowing through this function and escaping without being relocated?

Reply:
> So how do we know that other things that need this kind of relocation aren't flowing through this function and escaping without being relocated?

Not sure we can know that with the design we have at the moment. As long as Instruction Selection can generate instructions that aren't directly related to the node/symref that was being evaluated we're going to need something like a white list for later phases to finish doing what the evaluator started.